src/main/java/io/confluent/connect/jdbc/sink/PreparedStatementBinder.java

```diff
@@ -115,7 +142,13 @@ protected int bindKeyFields(SinkRecord record, int index) throws SQLException {
       case RECORD_VALUE: {
         for (String fieldName : fieldsMetadata.keyFieldNames) {
           final Field field = schemaPair.valueSchema.field(fieldName);
-          bindField(index++, field.schema(), ((Struct) record.value()).get(field));
+          if (tabDef == null) {
+            bindField(index++, field.schema(), ((Struct) record.value()).get(field));
```

Comment:
We should only call the new method, and let it delegate to the old method as a fallback.
When we don't have a table definition, we can pass-through the `null`.
Also, move this logic into the local `bindField` method and pass in the fieldName to deduplicate this logic.